apps/jobs/views.py

```diff
@@ -1027,7 +1044,25 @@ def re_run_submission(request, submission_pk):
         }
         return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
 
-    publish_submission_message(challenge.pk, challenge_phase.pk, submission.pk)
+    message = {
+        "challenge_pk": challenge.pk,
+        "phase_pk": challenge_phase.pk,
+        "submission_pk": submission.pk
+    }
+
+    if submission.challenge_phase.challenge.is_docker_based:
+        try:
+            response = requests.get(submission.input_file)
+        except Exception as e:
+            response_data = {
+                "error": "Failed to get submission input file with error: {0}".format(e)
+            }
+            return Response(response_data, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
```

Comment:
The error code here should be 404 NOT FOUND.

Reply:
If response for `re_run_submission` is `404`, it might mean something the browser/client was not couldn't endpoint. I kept here 500 endpoint is further using http calls, which might return 40x errors.
Please let me know if it makes sense.

Reply:
I see..sure!